tests/unit/challenges/test_views.py

```diff
@@ -3643,3 +3643,53 @@ def test_star_challenge(self):
         response = self.client.post(self.url, {})
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class GetChallengePhaseByPk(BaseChallengePhaseClass):
+
+    def setUp(self):
+        super(GetChallengePhaseByPk, self).setUp()
+        self.url = reverse_lazy('challenges:get_challenge_phase_by_pk',
+                                kwargs={'pk': self.challenge_phase.pk})
+
+    def test_get_challenge_phase_by_pk(self):
+        expected = {
+            "id": self.challenge_phase.id,
+            "name": self.challenge_phase.name,
+            "description": self.challenge_phase.description,
+            "leaderboard_public": self.challenge_phase.leaderboard_public,
+            "start_date": "{0}{1}".format(self.challenge_phase.start_date.isoformat(), 'Z').replace("+00:00", ""),
+            "end_date": "{0}{1}".format(self.challenge_phase.end_date.isoformat(), 'Z').replace("+00:00", ""),
+            "challenge": self.challenge_phase.challenge.pk,
+            "max_submissions_per_day": self.challenge_phase.max_submissions_per_day,
+            'max_submissions_per_month': self.challenge_phase.max_submissions_per_month,
+            "max_submissions": self.challenge_phase.max_submissions,
+            "is_public": self.challenge_phase.is_public,
+            "is_active": True,
+            "codename": self.challenge_phase.codename,
+            "slug": self.challenge_phase.slug,
+        }
+        response = self.client.get(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_challenge_phase_by_pk_when_user_is_not_authenticated(self):
```

Comment:
Hey, @Sanji515 I just noticed. The API [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/views.py#L1868) doesn't need authentication. Hence this test case isn't required. Moreover, please add the test case for this endpoint URL also.

Reply:
Hi @RishabhJain2018, I don't get it actually, of which endpoint URL you want me to write tests?

Reply:
This URL: https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/urls.py#L144